pkg/ccl/changefeedccl/changefeedbase/settings.go

```diff
@@ -356,3 +356,15 @@ var Quantize = settings.RegisterDurationSettingWithExplicitUnit(
 	0,
 	settings.NonNegativeDuration,
 )
+
+// KafkaV2IncludeErrorDetails enables detailed error messages for Kafka v2 sinks
+// when message_too_large errors occur. This includes the message key, size,
+// and MVCC timestamp in the error.
+var KafkaV2IncludeErrorDetails = settings.RegisterBoolSetting(
```

Comment:
I think it may be better if you make a PR that adds this cluster setting to master and then we backport that to 25.2. The setting will need to be on master anyway (either active like in this PR or retired).